src/State.hs

```diff
@@ -280,12 +281,21 @@ updatePreference pref = do
 -- occurs.
 refreshChannelsAndUsers :: MH ()
 refreshChannelsAndUsers = do
+  -- The below code is a duplicate of mmGetAllChannelsWithDataForUser function,
+  -- which has been inlined here to gain a concurrency benefit.
   session <- use (csResources.crSession)
   myTeamId <- use (csMyTeam.teamIdL)
   myId <- use (csMe.userIdL)
   doAsyncWith Preempt $ do
-    chansWithData <- mmGetAllChannelsWithDataForUser session myTeamId myId
-    uMap <- mmGetProfiles session myTeamId 0 10000
+    (chans, datas, uMap) <- runConcurrently $ (,,)
```

Comment:
As a minor note, adding concurrency "under" the work queue like this (which is well done here) will cause incorrect reporting of background async work counts (https://github.com/matterhorn-chat/matterhorn/blob/master/src/Draw/Main.hs#L585).  That's not really an issue for this change, but something to keep in mind if more of this style is used.  

The background work counts are more of a debugging aid anyhow, but the utility will be less if they become consistently inaccurate.  I don't think it's worth adding *significant* complexity to maintain the accuracy; just something to make a deliberate decision on if more of this concurrency style is introduced in the future.

That said, I like this change, thank you @abhin4v !

Reply:
Thanks, @kquick - I thought the work count was the queue length, and this doesn't change that, right? How does it become inaccurate?

Reply:
Right, the work count is the queue length, but as I understand this PR, this single work queue item will end up issuing 3 requests to the server.  Previously there was a 1:1 correlation between queue entry and server request, so I suppose it depends on what the "work count" means, but it's probably most useful as a general indicator of pending server interactions (i.e. the "slow" stuff).

Reply:
> Previously there was a 1:1 correlation between queue entry and server request

That isn't the case; there are quite a few async work queue items that make multiple (serial) requests to the server in the course of their work.

Reply:
Ahh, right.  And I didn't have to look very far to find one and remember the pattern: https://github.com/matterhorn-chat/matterhorn/blob/master/src/State.hs#L227-L232.  I updated the description to keep myself from getting confused again.  :-)  https://github.com/matterhorn-chat/matterhorn/commit/7fce525d9c6c4a8a50f491a8e2088bfa59c4311f